### DISCUSSION TITLE
3rd party scripts cause CORS errors

### DISCUSSION THREAD

**Discussion by MushuPlatypus** [vercel/next.js#38516] [Help]
When I add a script via Google Tagmanager I get a CORS error if that script tries to fetch some settings.  The same happens when I try to use the Script tag.

Here I'm loading this script: https://apps.elfsight.com/p/platform.js which then tries to fetch the settings.
`Access to XMLHttpRequest at 'https://apps.elfsight.com/p/boot/?w=7f438eaa-105b-4e16-a44d-7e1b16e973c4' from origin 'http://aau.local:3001' has been blocked by CORS policy: Request header field traceparent is not allowed by Access-Control-Allow-Headers in preflight response.`

**Comment by EdPutans** [comment]
Setting this header in `next.config.js` helped
```
    source: '/:path*',
    headers: [
        {
            key: 'Content-Security-Policy',
            value: `frame-ancestors 'self' ${domain};`,
        },
    ],
    ```
    
                More here: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/frame-ancestors
